kata02/chop.go

```diff
@@ -1,6 +1,52 @@
 package kata02
 
-//Chop1 is my first chop method
+type Chop func(int, []int) int
+
 func Chop1(needle int, haystack []int) int {
-	return -2
+	haystackLength := len(haystack)
+	switch haystackLength {
+	case 0:
+		return -1
+	case 1:
+		if haystack[0] == needle {
+			return 0
+		}
+		return -1
+	default:
+		half1 := haystack[:haystackLength/2]
+		half2 := haystack[haystackLength/2:]
+		inHalf1 := Chop1(needle, half1)
+		if inHalf1 >= 0 {
+			return inHalf1
+		}
+		inHalf2 := Chop1(needle, half2)
+		if inHalf2 >= 0 {
+			return haystackLength/2 + inHalf2
+		}
+		return -1
+	}
```

Comment:
my first go method on my own... fairly simple recursion.  I know there is some cleanup that can be done here already, but leaving as-is for the moment.